src/main/java/bio/terra/service/dataset/flight/create/CreateDatasetAuthzIamPolicyStep.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.service.dataset.flight.create;
+
+import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.service.iam.IamProviderInterface;
+import bio.terra.service.iam.IamRole;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.UUID;
+
+public class CreateDatasetAuthzIamPolicyStep implements Step {
+    private static final Logger logger = LoggerFactory.getLogger(CreateDatasetAuthzIamPolicyStep.class);
+
+    private final IamProviderInterface iamClient;
+    private final AuthenticatedUserRequest userReq;
+
+    public CreateDatasetAuthzIamPolicyStep(
+        IamProviderInterface iamClient,
+        AuthenticatedUserRequest userReq) {
+        this.iamClient = iamClient;
+        this.userReq = userReq;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) throws InterruptedException {
+        FlightMap workingMap = context.getWorkingMap();
+        UUID datasetId = workingMap.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
+        Map<IamRole, String> policyEmails = iamClient.syncDatasetResourcePolicies(userReq, datasetId);
+        workingMap.put(DatasetWorkingMapKeys.POLICY_EMAILS, policyEmails);
+        return StepResult.getStepResultSuccess();
+    }
+
+    @Override
+    public StepResult undoStep(FlightContext context) throws InterruptedException {
+        return StepResult.getStepResultSuccess();
```

Comment:
I'm not sure but is there anything we could do to clean up here in case something goes wrong while syncing?

Reply:
I guess it would probably happen when we delete the resource while rolling back

Reply:
Ok, after looking into it a bit more - I don't think the "syncOnePolicy" call actually makes any changes. It appears to be a mechanism to retrieve the policy email.  So, it makes sense that we don't "undo" anything. 

But, I think this means that it doesn't need to be in its own step. The only reason to keep them in separate steps would be if we wanted to add a stairway retry step setting on top of the SAM retry call, but that seems like overkill.